apis/UserSettings/IUserSettings.h

```diff
@@ -331,6 +332,66 @@ struct EXTERNAL IUserSettings : virtual public Core::IUnknown
 
 };
 
+/**
+ * If a setting was owned by an other component prior to being migrated into UserSettings,
+ * we require the ability to detect when this migration has completed.
+ * The component which was the previous owner of a given setting has the responsibility
+ * to always set the setting on UserSettings interface (During migration AND during first time install).
+ * Until this is done the setting is not considered valid, meaning the the MigrationState will contain requiresMigration=true
+ * All settings not requiring migration would immediately be valid.
+ */
+
+/* @json 1.0.0 @text:keep */
+struct EXTERNAL IUserSettingsInspector : virtual public Core::IUnknown
+{
+    enum { ID = ID_USER_SETTINGS_INSPECTOR };
+
+    ~IUserSettingsInspector() override = default;
+
+    enum SettingsKey : uint32_t
+    {
+        PREFERRED_AUDIO_LANGUAGES = 1           /* @text PREFERRED_AUDIO_LANGUAGES */ ,
+        AUDIO_DESCRIPTION = 2                   /* @text AUDIO_DESCRIPTION */,
+        CAPTIONS = 3                            /* @text CAPTIONS */,
+        PREFERRED_CAPTIONS_LANGUAGES = 4        /* @text PREFERRED_CAPTIONS_LANGUAGES */,
+        PREFERRED_CLOSED_CAPTION_SERVICE = 5    /* @text PREFERRED_CLOSED_CAPTION_SERVICE */,
+        PRESENTATION_LANGUAGE = 6               /* @text PRESENTATION_LANGUAGE */,
+        HIGH_CONTRAST = 7                       /* @text HIGH_CONTRAST */,
+        PIN_CONTROL = 8                         /* @text PIN_CONTROL */,
+        VIEWING_RESTRICTIONS = 9                /* @text VIEWING_RESTRICTIONS */,
+        VIEWING_RESTRICTIONS_WINDOW = 10        /* @text VIEWING_RESTRICTIONS_WINDOW */,
+        LIVE_WATERSHED = 11                     /* @text LIVE_WATERSHED */,
+        PLAYBACK_WATERSHED = 12                 /* @text PLAYBACK_WATERSHED */,
+        BLOCK_NOT_RATED_CONTENT = 13            /* @text BLOCK_NOT_RATED_CONTENT */,
+        PIN_ON_PURCHASE = 14                    /* @text PIN_ON_PURCHASE */,
+        VOICE_GUIDANCE = 15                     /* @text VOICE_GUIDANCE */,
+        VOICE_GUIDANCE_RATE = 16                /* @text VOICE_GUIDANCE_RATE */,
+        VOICE_GUIDANCE_HINTS = 17               /* @text VOICE_GUIDANCE_HINTS */
+    };
+
+    struct SettingsMigrationState
+    {
+        SettingsKey key             /* @text key */ ;
+        bool requiresMigration      /* @text requiresMigration */;
+    };
+
+    using IUserSettingsMigrationStateIterator = RPC::IIteratorType<SettingsMigrationState, ID_USER_SETTINGS_MIGRATION_STATE_ITERATOR>;
+
+    /** Get the migration state of the respective key */
+    // @text getMigrationState
+    // @brief Get the migration state of the respective key
+    // @param key: one of UserSettingsKey
+    // @param migrationState: key and it's migration state.
+    virtual Core::hresult GetMigrationState(const SettingsKey key, SettingsMigrationState &migrationState/* @out */) const = 0;
```

Comment:
This is not in sync with API Spec https://etwiki.sys.comcast.net/display/RDKV/User+Settings+Inspector

The result object should have a param "requiresMigration" and nothing else.